dev/checkstyle-suppressions.xml

```diff
@@ -30,6 +30,8 @@
 <suppressions>
     <suppress checks=".*"
               files="core/src/main/java/org/apache/spark/util/collection/TimSort.java"/>
+    <suppress checks="NoFinalizer"
+              files="core/src/main/java/org/apache/spark/io/NioBufferedFileInputStream.java"/>
```

Comment:
@HyukjinKwon  Also we could try `// scalastyle:off ...  // scalastyle:on`. I'll test this way.